astropy/utils/masked/tests/test_function_helpers.py

```diff
@@ -1453,6 +1453,185 @@ def test_nanpercentile(self):
         self.check(np.nanpercentile, q=50)
 
 
+class TestArraySetOps:
+    """Tests based on those from numpy.ma.tests.test_extras.
+
+    Adjusted to take into account that comparing masked values should
+    result in masked equality.
+
+    """
+
+    @classmethod
+    def setup_class(self):
```

Comment:
nit, but `@classmethod` means that the first argument isn't an instance, so `cls` would be more appropriate than `self`... that said, it's conform to how the rest of the module is doing it, and it actually works as you'd expect, so it's probably not worth fixing it here (but I'm tempted to propose this as a ruff rule so it can be trivially fixed everywhere all the time)

Reply:
This is really a left-over from when `pytest` did not treat these as class methods. The only advantage  to keeping `self` was that in the tests one refers to `self.x` everywhere too.

But indeed it would not be bad to just change it throughout.

Reply:
I'm making a note to self to create a feature request to ruff then. Meanwhile, feel free to resolve this thread. Thanks !

Reply:
The rule already exists, we're just not enforcing it: https://github.com/astropy/astropy/blob/286948c761cfc71925305e780b8184fd0e9e4059/.ruff.toml#L90

Reply:
oh, thank you for pointing it out !